app/helpers/foreman_ansible/ansible_roles_helper.rb

```diff
@@ -38,5 +38,18 @@ def import_time(role)
     def roles_attrs(roles)
       roles.map { |item| { :id => item.id, :name => item.name } }
     end
+
+    def model_roles_attrs(form_object, assoc_class, foreign_key, assoc_key)
+      inherited_attrs = roles_attrs form_object.inherited_ansible_roles_ordered
+
+      own_attrs = assoc_class.includes(:ansible_role).
```

Comment:
in Puppet we have similar issue and it's solved by both Host and Hostgroup to respond to `all_ansible_roles_ordered` and then we just iterate over those. Would something like that be possible, or we need the distinction where the roles came from in the result?

Reply:
We need to know if the role is inherited or not.